test/common/network/address_impl_test.cc

```diff
@@ -177,9 +202,26 @@ TEST(Ipv6InstanceTest, BadAddress) {
 }
 
 TEST(Ipv6InstanceTest, SocketBindAndConnect) {
-  // Test listening on and connecting to an unused port on the IPv4 loopback address.
-  Ipv6Instance loopbackPort("::1", ::Network::Test::getUnusedPort());
-  testSocketBindAndConnect(loopbackPort);
+  // Test listening on and connecting to an unused port on the IPv6 loopback address.
+  testSocketBindAndConnect("[::1]:0");
+}
+
+TEST(Ipv6InstanceTest, ParseInternetAddressAndPort) {
+  EXPECT_EQ(nullptr, parseInternetAddressAndPort("::1"));
+  EXPECT_EQ(nullptr, parseInternetAddressAndPort("::"));
+  EXPECT_EQ(nullptr, parseInternetAddressAndPort("[::]:1]:2"));
```

Comment:
There's probably a bunch of other improperly nested permutations to consider. Even if we have full code coverage with what's here I would prefer to see some of these, e.g. "]::[:]" etc.